File: Coiny.API/Models/Transaction.cs
namespace Coiny.API.Models;

public class Transaction : BaseEntity
{
    public int HouseholdId { get; set; }
    public Household Household { get; set; } = null!;
    public int AccountId { get; set; }
    public Account Account { get; set; } = null!;
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public decimal Amount { get; set; }
    public DateTime TransactionDate { get; set; }
    public string Description { get; set; } = string.Empty;
    public string? Merchant { get; set; }
    public string? Notes { get; set; }
    public bool IsCleared { get; set; }
}

File: Coiny.Web/src/app/shared/models/transactions/transaction.ts
export interface Transaction { 
    id: number;
    accountName: string;
    institutionName: string | null;
    categoryName: string;
    categoryType: string;
    amount: number;
    transactionDate: string;
    description: string;
    merchant: string | null;
    isCleared: boolean;
}